File: tamer/python/cat.py
#!/usr/bin/env python3

import sys

def cat_by_line(src):
    for line in src:
        print(line, end="")

def main(argc, argv):
    if argc > 1:
        for fname in argv[1:]:
            try:
                with open(fname, 'r') as src:
                    cat_by_line(src)
            except FileNotFoundError:
                pass
    else:
        cat_by_line(sys.stdin)

    return 0



###################################################################################################
if __name__ == "__main__":
    sys.exit(main(len(sys.argv), sys.argv))

File: tamer/c/cat.cpp
#include <iostream>
#include <fstream>
#include <string>

/*************************************************************************************************/
void cat_by_line(std::istream& in) {
    std::string line;

    while (std::getline(in, line)) {
        std::cout << line << std::endl;
    }
}


/*************************************************************************************************/
int main(int argc, char* argv[]) {
    if (argc > 1) {
        for (int idx = 1; idx < argc; idx++) {
            std::ifstream src(argv[idx]);

            if (src.is_open()) {
                cat_by_line(src);
                src.close();
            }
        }
    } else {
        cat_by_line(std::cin);
    }

    return 0;
}
